src/libsyntax/parse/token.rs

```diff
@@ -48,6 +48,8 @@ pub enum DelimToken {
     Bracket,
     /// A curly brace: `{` or `}`
     Brace,
+    /// An empty delimiter
+    Empty,
```

Comment:
On a side note, if you wanted to do some refactoring, I'd love to rename `DelimToken` to `Delimiter` and stop importing the variants and instead use, e.g., `Delimter::Brace` rather than `token::Brace` throughout the codebase. (To be clear, not in this PR though)